### DISCUSSION TITLE
extract `Sequence` as `PyType`?

### DISCUSSION THREAD

**Discussion by samuelcolvin** [PyO3/pyo3#2693] [Questions]
I'd like to do an `is_instance` check with `typing.Sequence`, however

```rs
let sequence_type: &PyType = sequence_obj.extract();
```

Is failing with `TypeError: '_SpecialGenericAlias' object cannot be converted to 'PyType'`.

Is this a fundamental limitation of the python C API, or something that pyo3 could support?

Mostly for my record, here's the failing test from [pydantic-core](https://github.com/pydantic/pydantic-core):

```py
def test_is_instance_sequence():
    v = SchemaValidator({'type': 'is-instance', 'cls': typing.Sequence})
    assert v.isinstance_python(1) is False
    assert v.isinstance_python([1]) is True
```

Fails when creating the `SchemaValidator` with:

```
    def test_is_instance_sequence():
>       v = SchemaValidator({'type': 'is-instance', 'cls': typing.Sequence})
E       pydantic_core._pydantic_core.SchemaError: Error building "is-instance" validator:
E         TypeError: '_SpecialGenericAlias' object cannot be converted to 'PyType'

tests/validators/test_is_instance.py:204: SchemaError
```

**Comment by birkenfeld** [comment]
OK, I assume `sequence_obj is typing.Sequence`. This is a case not supported by our current API, but without problems in the C API.

**Comment by birkenfeld** [comment]
I've opened #2694 to track this.
